pytest_twisted.py

```diff
@@ -208,7 +208,15 @@ def init_twisted_greenlet():
 
 def stop_twisted_greenlet():
     if _instances.gr_twisted:
-        _instances.reactor.stop()
+        # check if the reactor is not stopped first to prevent calling stop() twice which results
+        # into an error, traceback printed and process hanging until it is killed with SIGKILL.
+        # This sometimes happens when the pytest process is sent a SIGINT signal.
+        # The reactor will stop itself as it register a SIGINT signal for itself and then it is
+        # stopped second time here.
+        # I don't understand when exactly is this happening but I am also observing here that
+        # reactor.running == True and reactor._stopped == True.
+        if not _instances.reactor._stopped:
+            _instances.reactor.stop()
```

Comment:
[`._stopped` is set](https://github.com/twisted/twisted/blob/e0c069b429a66343e3e6c7ef19df512710f4775c/src/twisted/internet/base.py#L698) to request termination of the reactor.  I didn't dig it up, but I presume that somewhere else the reactor will set `.running` to `False`.  The suggestion below avoids accessing the private `._stopped` member.

Does this happen often enough that you can run the changed code a few times and be confident it has addressed the issue?

```suggestion
        try:
            _instances.reactor.stop()
        except error.ReactorNotRunning:
            # Sometimes the reactor is stopped before we get here.  For example, this can
            # happen in response to a SIGINT in some cases.
            pass
```

Reply:
> ._stopped is set to request termination of the reactor. I didn't dig it up, but I presume that somewhere else the reactor will set .running to False. 

same.

>  The suggestion below avoids accessing the private ._stopped member.

good idea.

> Does this happen often enough that you can run the changed code a few times and be confident it has addressed the issue?

It's random but hapens frequently to me.

I believe the patch you suggested fixes the issue based on both review and some non-rigorous testing.